server/db/interface.go

```diff
@@ -59,6 +59,17 @@ type OrderStatus struct {
 	Fill   uint64
 }
 
+// PreimageResult is the outcome of preimage collection for an order in an epoch
+// that closed at a certain time.
+type PreimageResult struct {
+	Miss bool
+	Time int64
+	// NOTE: There can be multiple results with the same time (same epoch) but
+	// since there is unlikely to be a mix of misses and successes in the same
+	// epoch, the sorting ambiguity is acceptable. If that proves to be a
+	// problem, order ID can be added to this struct.
```

Comment:
Your comments on this are correct, and I'm planning to address it in https://github.com/decred/dcrdex/issues/712 because I'm going to need to add `value` or `quantity` to the match outcomes.  While that is mainly for match outcomes, I can add order ID to the PreimageResults as well.